snuba/datasets/storage.py

```diff
@@ -0,0 +1,91 @@
+from abc import ABC
+from typing import Optional, Sequence
+
+
+from snuba.datasets.dataset_schemas import StorageSchemas
+from snuba.datasets.table_storage import TableWriter
+from snuba.query.query import Query
+from snuba.query.query_processor import QueryProcessor
+from snuba.request.request_settings import RequestSettings
+
+
+class Storage(ABC):
+    """
+    Storage is an abstraction on anything we can run a query onto in our
+    database. This means that it generally represents a Clickhouse table
+    or a view.
+    It provides:
+    - what we need to build the query (the schemas)
+    - it can provide a table writer if we can write on this storage
+    - a sequence of query processors that are applied to the query after
+      the storage is selected.
+
+    There are one or multiple storages per dataset (in the future, there
+    will be multiple per entity). During the query processing a storage
```

Comment:
What is entity here?

Reply:
There is a description in the PR description